src/client/datagram.cc

```diff
@@ -59,43 +59,43 @@ void DatagramDestination::SendDatagramTo(
     uint16_t fromPort,
     uint16_t toPort) {
   uint8_t buf[MAX_DATAGRAM_SIZE];
-  auto identityLen =
-    m_Owner.GetIdentity().ToBuffer(buf, MAX_DATAGRAM_SIZE);
+  auto identityLen = m_Owner.GetIdentity().ToBuffer(buf, MAX_DATAGRAM_SIZE);
   uint8_t* signature = buf + identityLen;
   auto signatureLen = m_Owner.GetIdentity().GetSignatureLen();
   uint8_t* buf1 = signature + signatureLen;
   size_t headerLen = identityLen + signatureLen;
   memcpy(buf1, payload, len);
-  if (m_Owner.GetIdentity().GetSigningKeyType() ==
-      i2p::data::SIGNING_KEY_TYPE_DSA_SHA1) {
+  if (m_Owner.GetIdentity().GetSigningKeyType()
+      == i2p::data::SIGNING_KEY_TYPE_DSA_SHA1) {
     uint8_t hash[32];
     i2p::crypto::SHA256().CalculateDigest(hash, buf1, len);
     m_Owner.Sign(hash, 32, signature);
   } else {
     m_Owner.Sign(buf1, len, signature);
   }
-  auto msg =
-    CreateDataMessage(buf, len + headerLen, fromPort, toPort);
-  auto remote = m_Owner.FindLeaseSet(ident);
-  if (remote)
+  I2NPMessage* msg = CreateDataMessage(buf, len + headerLen, fromPort, toPort);
+  std::shared_ptr<const i2p::data::LeaseSet> remote
+      = m_Owner.FindLeaseSet(ident);
```

Comment:
I don't see a need to refactor-out ```auto``` (to do so is counterproductive for abstraction and should be refactored back).